File: ZoneSentry/ClientApp/src/api/models/ApplicationUser.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ApplicationUserType } from './ApplicationUserType';
import type { ConstructionCompany } from './ConstructionCompany';
import type { Realty } from './Realty';

export type ApplicationUser = {
    type: ApplicationUserType;
    firstName?: string | null;
    lastName?: string | null;
    patronymic?: string | null;
    ownedRealties?: Array<Realty> | null;
    constructionCompany?: ConstructionCompany;
    id?: number;
    userName?: string | null;
    normalizedUserName?: string | null;
    email?: string | null;
    normalizedEmail?: string | null;
    emailConfirmed?: boolean;
    passwordHash?: string | null;
    securityStamp?: string | null;
    concurrencyStamp?: string | null;
    phoneNumber?: string | null;
    phoneNumberConfirmed?: boolean;
    twoFactorEnabled?: boolean;
    lockoutEnd?: string | null;
    lockoutEnabled?: boolean;
    accessFailedCount?: number;
};

File: ZoneSentry/Models/ApplicationUser.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Identity;

namespace ZoneSentry.Models;

public class ApplicationUser : IdentityUser<int>
{
    [Required]
    public ApplicationUserType Type { get; set; }
    
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string? Patronymic { get; set; }
    
    public List<Realty> OwnedRealties { get; set; }

    public ConstructionCompany? ConstructionCompany { get; set; }
}

public enum ApplicationUserType
{
    User,
    ConstructionCompanyAdmin,
    ServiceProvider
}

public class ApplicationUserDTO {
    public int Id { get; set; }
    public string UserName { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string? Patronymic { get; set; }
    public string Email { get; set; }
    [Required]
    public ApplicationUserType Type { get; set; }
}